explorer-ui/src/components/ProfileView/Preferences/Preferences.jsx

```diff
@@ -0,0 +1,273 @@
+      occupation: preferenceInfo.occupation,
+      budget: preferenceInfo.budget,
+      currLocation: preferenceInfo.currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    console.log("preferenceInfo: ", preferenceInfo);
+    console.log("attractions so far: ", touristAttractions);
+  }
+
+  // Update the state array for the users attractions interests
+  function handleOccupationsSelected(event, occupation) {
+    event.preventDefault();
+    setEmployment(occupation.current.value);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: preferenceInfo.attractions,
+      occupation: employment,
+      budget: preferenceInfo.budget,
+      currLocation: preferenceInfo.currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    console.log("occupation is: ", employment);
+  }
+
+  // Update the state array for the users attractions interests
+  function handleBudgetSelected(event, selectedBudget) {
+    event.preventDefault();
+    setBudget(selectedBudget.current.value);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: preferenceInfo.attractions,
+      occupation: preferenceInfo.occupation,
+      budget: budget,
+      currLocation: preferenceInfo.currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    console.log("budget is: ", budget);
+  }
+
+  // Update the state array for the users attractions interests
+  function handleVisibilitySelected(event, visibililityChoice) {
+    event.preventDefault();
+    setVisibility(visibililityChoice.current.value);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: preferenceInfo.attractions,
+      occupation: preferenceInfo.occupation,
+      budget: preferenceInfo.budget,
+      currLocation: preferenceInfo.currLocation,
+      visibility: visibility,
+    });
+    console.log("visibility is: ", visibililityChoice);
+  }
+  // Update the state array for the users attractions interests
+  function handleLocationSet(event, region) {
+    event.preventDefault();
+    setCurrLocation(region.current.value);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: preferenceInfo.attractions,
+      occupation: preferenceInfo.employment,
+      budget: preferenceInfo.budget,
+      currLocation: currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    // Get latitude & longitude from address.
+    Geocode.fromAddress(region.current.value).then(
+      (response) => {
+        const { lat, lng } = response.results[0].geometry.location;
+        setPosition({ lat: lat, lng: lng });
+        console.log(lat, lng);
+      },
+      (error) => {
+        console.error(error);
+      }
+    );
+
+    console.log("current location is: ", currLocation);
+  }
+
+// Send the profile information to the database
+  function handleSubmit(event) {
+    console.log("submitted");
+    event.preventDefault();
```

Comment:
nit - generally will not need preventDefault unless its a native browser event such as form tag submit